src/connector/src/common.rs

```diff
@@ -604,13 +608,28 @@ impl NatsCommon {
     > {
         let context = self.build_context().await?;
         let stream = self.build_or_get_stream(context.clone(), stream).await?;
-        let subject_name = self
-            .subject
-            .replace(',', "-")
-            .replace(['.', '>', '*', ' ', '\t'], "_");
+        let subject_name: String = match self.filter_subject {
+            Some(ref v) => v.replace(',', "-").replace(['.', '>', '*', ' ', '\t'], "_"),
+            None => String::new(),
+        };
         let name = format!("risingwave-consumer-{}-{}", subject_name, split_id);
+
```

Comment:
If user create 2 nats connectors with `self.fileter_subject` is None, then the consumer name will be the same. Use the same name to build nats consumer, need to confirm whether it will cause problem.

Reply:
Yea this may cause issues since its not making a new stream I can change this to be a user defined `consumer_name` in the config. Should solve edge cases